### ISSUE TITLE
Fix user-skills and user-categories index actions to return authenticated user data

### ISSUE THREAD

**Issue by joshsmith** [code-corps/code-corps-api#204]
Right now the `index` actions in both `user-skills` and `user-categories` return everyone's data, instead of the authenticated user's when authenticated.

In Rails we did the following:

```
    if id_params.present?
      user_skills = UserSkill.includes(:skill, :user).where(id: id_params)
    elsif current_user.present?
      user_skills = UserSkill.includes(:skill, :user).where(user_id: current_user.id)
    else
      user_skills = []
    end
```

We need to fix this the `filter` or find some other approach to correcting the data returned from this endpoint.

**Comment by joshsmith** [comment]
For #26.

**Comment by joshsmith** [comment]
Closed by #216
